chain/chain/src/stateless_validation/chunk_validation.rs

```diff
@@ -580,16 +597,22 @@ pub fn validate_chunk_state_witness(
         .with_label_values(&[&state_witness.chunk_header.shard_id().to_string()])
         .start_timer();
     let span = tracing::debug_span!(target: "client", "validate_chunk_state_witness").entered();
+    let witness_shard_layout = epoch_manager.get_shard_layout(&state_witness.epoch_id)?;
+    let witness_chunk_shard_id = state_witness.chunk_header.shard_id();
+    let witness_chunk_shard_uid =
+        epoch_manager.shard_id_to_uid(witness_chunk_shard_id, &state_witness.epoch_id)?;
     let block_hash = pre_validation_output.main_transition_params.block_hash();
     let epoch_id = epoch_manager.get_epoch_id(&block_hash)?;
```

Comment:
It's a bit annoying that we have two different epoch ids - the `state_witness.epoch_id` and this one. I don't have a good suggestion, just ranting.